cmd/query/app/grpc_handler.go

```diff
@@ -177,3 +190,123 @@ func (g *GRPCHandler) GetDependencies(ctx context.Context, r *api_v2.GetDependen
+	// Check for cases where clients do not provide the Quantile, which defaults to the float64's zero value.
+	if r.Quantile == 0 {
+		return nil, errMissingQuantile
+	}
+	bqp, err := g.newBaseQueryParameters(r.BaseRequest)
+	if err := g.handleErr("failed to build parameters", err); err != nil {
+		return nil, err
+	}
+	queryParams := metricsstore.LatenciesQueryParameters{
+		BaseQueryParameters: bqp,
+		Quantile:            r.Quantile,
+	}
+	m, err := g.metricsQueryService.GetLatencies(ctx, &queryParams)
+	if err := g.handleErr("failed to fetch latencies", err); err != nil {
+		return nil, err
+	}
+	return &metrics.GetMetricsResponse{Metrics: *m}, nil
+}
+
+// GetCallRates is the gRPC handler to fetch call rate metrics.
+func (g *GRPCHandler) GetCallRates(ctx context.Context, r *metrics.GetCallRatesRequest) (*metrics.GetMetricsResponse, error) {
+	bqp, err := g.newBaseQueryParameters(r.BaseRequest)
+	if err := g.handleErr("failed to build parameters", err); err != nil {
+		return nil, err
+	}
+	queryParams := metricsstore.CallRateQueryParameters{
+		BaseQueryParameters: bqp,
+	}
+	m, err := g.metricsQueryService.GetCallRates(ctx, &queryParams)
+	if err := g.handleErr("failed to fetch call rates", err); err != nil {
+		return nil, err
+	}
+	return &metrics.GetMetricsResponse{Metrics: *m}, nil
+}
+
+// GetErrorRates is the gRPC handler to fetch error rate metrics.
+func (g *GRPCHandler) GetErrorRates(ctx context.Context, r *metrics.GetErrorRatesRequest) (*metrics.GetMetricsResponse, error) {
+	bqp, err := g.newBaseQueryParameters(r.BaseRequest)
+	if err := g.handleErr("failed to build parameters", err); err != nil {
+		return nil, err
+	}
+	queryParams := metricsstore.ErrorRateQueryParameters{
+		BaseQueryParameters: bqp,
+	}
+	m, err := g.metricsQueryService.GetErrorRates(ctx, &queryParams)
+	if err := g.handleErr("failed to fetch error rates", err); err != nil {
+		return nil, err
+	}
+	return &metrics.GetMetricsResponse{Metrics: *m}, nil
+}
+
+// GetMinStepDuration is the gRPC handler to fetch the minimum step duration supported by the underlying metrics store.
+func (g *GRPCHandler) GetMinStepDuration(ctx context.Context, _ *metrics.GetMinStepDurationRequest) (*metrics.GetMinStepDurationResponse, error) {
+	minStep, err := g.metricsQueryService.GetMinStepDuration(ctx, &metricsstore.MinStepDurationQueryParameters{})
+	if err := g.handleErr("failed to fetch min step duration", err); err != nil {
+		return nil, err
+	}
+	return &metrics.GetMinStepDurationResponse{MinStep: minStep}, nil
+}
+
+func (g *GRPCHandler) handleErr(msg string, err error) error {
+	if err == nil {
+		return nil
+	}
+	g.logger.Error(msg, zap.Error(err))
+
+	// Avoid wrapping "expected" errors with an "Internal Server" error.
+	switch {
+	case errors.Is(err, disabled.ErrDisabled):
+		return errGRPCMetricsQueryDisabled
+	case errors.Is(err, errMissingServiceNames), errors.Is(err, errMissingQuantile):
```

Comment:
rather than enumerating, could we check if it's already an instance of status.Error?